gitpod-io/gitpod
# Add ide-service server

## Description

Add ide-service to resolve all IDE related things. First implement to bring 
- Make `ide-service` a component and deploy
- Resolve image digest
- Resolve nightly/latest tag of image
- Verify config content
- API to get config content (string)

## Related Issue(s)

Relates https://github.com/gitpod-io/gitpod/issues/9189

## How to test

- Open this PR with Gitpod
- Exec commands below to start a test service
```
cd components/ide-service
go run main.go run --config ./example-config.json
```
- Exec test command to get parsed ide content
```
go run main.go test | jq
```
- `latestImage` should not contain `nightly` or `latest` tag, should be resolved

### ide-service in prev env
- Open this PR with Gitpod
- Wait until preview env installed
- Exec commands below

```
# exec sh
kubectl exec -it ide-service<press_tab_to_get> -c ide-service sh

# run test cmd
./app/ide-service test --config /config/config.json
```
- Output should be correct and align to server ide config response

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:
